### PR TITLE
[interp] Make newobj_vt[st]_fast non-recursive.

### PR DESCRIPTION
**Pull request by jaykrell**
[interp] Make newobj_vtst_fast and newobj_vt_fast not recursive.
    
This is builds upon https://github.com/mono/mono/pull/18875 which was newobj_vt_fast-only.
That was preceded by https://github.com/mono/mono/pull/18870 which did not handle the stack/GC perhaps as well.

Some notes:
 - This is a little gnarly in its stack manipulation in transform.c and interp.c. Please review carefully.
 - This is *not* known to contribute to WebAssembly/Blazor failures (which are suspected fixed via earlier fixes already). Leaving it alone is an option. Not my favorate, but I understand.
  - There are expected to be improvements on the Apple side as well, further strengthening the previous point. Still not my preference, but again I understand.

### CODE REVIEW

**Comment by jaykrell** [comment]
Ok I just need to update the title and description, and hope it works.

**Comment by jaykrell** [comment]
@monojenkins build failed

File: mono/mini/interp/interp.c [hunk 1]
```diff
@@ -5122,51 +5084,49 @@ call:;
...
-				interp_exec_method_newobj_vtst_fast (child_frame, context, error);
-
-				CHECK_RESUME_STATE (context);
-				sp->data.p = vt_sp;
-
+				// Put extra parameter and result on stack, before other parameters,
+				// and point stack to extra parameter, after result.
+				// This pattern occurs for newobj_vt_fast and newobj_fast.
+				sp [1].data.p = vt_sp;
+				sp [0].data.p = vt_sp;
```
Comment (jaykrell) [inline mono/mini/interp/interp.c]: Is writing return value before the call, here and elsewhere, worrisome, if the call throws an exception?
Previously it didn't work that way.

File: mono/mini/interp/transform.c [hunk 2]
```diff
@@ -4537,6 +4546,9 @@ generate_code (TransformData *td, MonoMethod *method, MonoMethodHeader *header,
 						// Set the method to be executed as part of newobj instruction
 						newobj_fast->data [0] = get_data_item_index (td, mono_interp_get_imethod (domain, m, error));
 					} else {
+						// Runtime inserts extra stack to hold return value, before call.
```
Comment (jaykrell) [inline mono/mini/interp/transform.c]: "runtime" means "interp.c", I should have said.

File: mono/mini/interp/interp.c (hunk 1 above)
Comment (BrzVlad) [reply mono/mini/interp/interp.c]: the contents of the stack here are not used by the exception handling mechanism and the values are lost anyway when catching in this frame or below.

**Comment by jaykrell** [comment]
@monojenkins build failed

**Comment by jaykrell** [comment]
@monojenkins build failed

**Pull request merged**